pkg/apis/core/validation/validation_test.go

```diff
@@ -2628,13 +2628,27 @@ func TestValidateVolumes(t *testing.T) {
 									FieldPath:  "metadata.labels",
 								},
 							},
+							{
+								Path: "labels with subscript",
+								FieldRef: &core.ObjectFieldSelector{
+									APIVersion: "v1",
+									FieldPath:  "metadata.labels['key']",
+								},
+							},
 							{
 								Path: "annotations",
 								FieldRef: &core.ObjectFieldSelector{
 									APIVersion: "v1",
 									FieldPath:  "metadata.annotations",
 								},
 							},
+							{
+								Path: "annotations with subscript",
+								FieldRef: &core.ObjectFieldSelector{
+									APIVersion: "v1",
+									FieldPath:  "metadata.annotations['key']",
+								},
+							},
```

Comment:
add a test case for a complex annotation name (`TEST.EXAMPLE.COM/key`)

Reply:
Done.